articles/142_idl.md

```diff
@@ -32,6 +32,14 @@ Further it describes how the interfaces are used to generate code in C, C++ and
 
 ROS 2 supports a subset of the [OMG IDL 4.2 specification](https://www.omg.org/spec/IDL/4.2).
 
+### Lexical Conventions
+
+#### 7.2.2 Comments
+
+#### 7.2.3 Identifiers
+
+#### 7.2.6 Literals
+
```

Comment:
Could a section be added here explicitly discussing the support or lack thereof for annotations? I'm not sure what the latest thinking is with respect to DDS's use of keyed fields/instances and whether or not RMW will ever support that concept, but of particular interest is whether or not annotations like @key fields might get preserved when generating vendor specific types. This would impact middleware behavior, even if ROS2 didn't explicitly provide support for the capabilities introduced, so it would be worth addressing either way.

Reply:
I added some information about annotations in 94a37507b0710a19540fcbdccbc0c79b48aff913.